packages/flutter/test/material/text_form_field_test.dart

```diff
@@ -531,4 +531,46 @@ void main() {
     final TextField widget = tester.widget(find.byType(TextField));
     expect(widget.selectionControls, equals(materialTextSelectionControls));
   });
+
+  testWidgets('TextFormField respects hintTextDirection', (WidgetTester tester) async {
+      await tester.pumpWidget(MaterialApp(
+        home: Material(
+          child: Directionality(
+            textDirection: TextDirection.rtl,
+            child: TextFormField(
+              decoration: InputDecoration(
+                border: OutlineInputBorder(),
+                labelText: 'Some Label',
+                hintText: 'Some Hint',
+                hintTextDirection: TextDirection.ltr,
+              ),
+            ),
+          ),
+        ),
+      ));
+
+      final hintTextFinder = find.text('Some Hint');
+
+      Text hintText = tester.firstWidget(hintTextFinder);
+      expect(hintText.textDirection, TextDirection.ltr);
```

Comment:
Why not `Directionality.of` for the first test?

Reply:
Because `Directionality.of` will be null here. Then it will default to App/OS's default directionality. This is just my idea, I will change it as you say.